app/components/container/containerController.js

```diff
@@ -117,22 +117,29 @@ function ($scope, $state, $stateParams, $filter, Container, ContainerCommit, Ima
   };
 
   $scope.confirmRemove = function () {
+    var title = 'You are about to remove a container.';
     if ($scope.container.State.Running) {
-      ModalService.confirmDeletion(
-        'You are about to remove a running container.',
-        function (confirmed) {
-          if(!confirmed) { return; }
-          $scope.remove();
+      title = 'You are about to remove a running container.';
+    }
+    //var text = '<label for="clean" class="control-label text-left">Clean associated volumes<portainer-tooltip position="bottom" message="Enable this option to remove associated volumes."></portainer-tooltip></label><label class="switch" style="margin-left: 20px;"><input type="checkbox"><i></i></label>';
+    var text = '<label for="clean" class="control-label text-left">Automatically clean associated volumes</label><label class="switch" style="margin-left: 21px;"><input type="checkbox"><i></i></label>';
+    ModalService.confirmContainerDeletion(
+      title,
+      text,
+        function (result) {
+          if(!result) { return; }
+          var cleanAssociatedVolumes = false;
+          for (var i in result) {
+            if (result[i] === 'on') cleanAssociatedVolumes = true;
+          }
+          $scope.remove(cleanAssociatedVolumes);
         }
       );
-    } else {
-      $scope.remove();
-    }
   };
 
-  $scope.remove = function() {
+  $scope.remove = function(cleanAssociatedVolumes) {
     $('#loadingViewSpinner').show();
-    Container.remove({id: $stateParams.id, force: true}, function (d) {
+    Container.remove({id: $stateParams.id, v: (cleanAssociatedVolumes) ? 1 : 0, force: true}, function (d) {
```

Comment:
What's the point of the ternary operation here? `cleanAssociatedVolumes` should contain a boolean value and will always be defined.